### PR TITLE
chore: react hooks eslint

### PR DESCRIPTION
**Pull request by imkesin**
## Description

The [react hooks plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) is more valuable and should be added at some point.

This PR gets to the point where all remaining issues are related to usage of `any`. These are much more difficult to resolve.

### CODE REVIEW

File: src/components/Button/CloseButton.tsx [hunk 1]
```diff
@@ -2,14 +2,12 @@ import React, { ButtonHTMLAttributes } from 'react'
 import CloseIcon from '../../icons/CloseIcon'
 import classNames from 'classnames'
 
-export interface CloseButtonProps
-  extends ButtonHTMLAttributes<HTMLButtonElement> {
+type CloseButtonProps = {
   textOnly?: boolean
-}
+} & Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'children'>
```
Comment (tomaszantas) [inline src/components/Button/CloseButton.tsx:7]: FYI: I had strange behavior when Prettier was enabled in the VSCode. After reloading Prettier, it gave other strange behavior. But I believe that we no longer use Prettier

Comment (tomaszantas) [reply src/components/Button/CloseButton.tsx:7]: I believe same change could be applied to `BackButton`

**Review by tomaszantas — APPROVED** [review]
LGTM!

File: src/components/Button/CloseButton.tsx (hunk 1 above)
Comment (imkesin) [reply src/components/Button/CloseButton.tsx:7]: Thank you for spotting that. I'll go change that component.

**Pull request merged**